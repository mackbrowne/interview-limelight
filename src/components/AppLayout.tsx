// Framework
import React from 'react';
import { Switch, Route } from 'react-router-dom';

import loadable from '@loadable/component';

// Components
import { Row, Col } from 'react-bootstrap';

import { AppContainer, Logo } from './AppLayout.style';

import logo from './../img/logo.svg';

const AsyncNavigation = loadable(() => import('./common/Navigation'));
const AsyncHome = loadable(() => import('./pages/Home'));
const AsyncNotFound = loadable(() => import('./pages/NotFound'));

const navigationItems = [{ label: 'Home', path: '/' }];

/**
 * Main rendering entry point for the app.
 *
 * @returns {Element} Stateless functional React component.
 */
const AppLayout = () => (
  <div className="App">
    <AppContainer>
      <Row className="bg-faded">
        <Col sm={2}>
          <Logo src={logo} alt="logo" className="img-fluid" fluid />
        </Col>
        <Col>
          <AsyncNavigation navigationItems={navigationItems} />
        </Col>
      </Row>
      <Switch>
        <Route exact path="/" component={AsyncHome} />
        <Route path="*" component={AsyncNotFound} />
      </Switch>
      <Row>
        <Col>
          <footer className="py-1 text-center">
            <p>
              Created by: <a href="https://mackbrowne.com/">Mack Browne</a>
            </p>
          </footer>
        </Col>
      </Row>
    </AppContainer>
  </div>
);

export default AppLayout;
