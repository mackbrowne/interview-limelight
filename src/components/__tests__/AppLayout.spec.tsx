import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

import AppLayout from '../AppLayout';

describe('<AppLayout /> Snapshot', () => {
  const render = () =>
    mount(
      <Router>
        <AppLayout />
      </Router>
    );
  it('renders snapshot with no props', () => {
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
  });
});
