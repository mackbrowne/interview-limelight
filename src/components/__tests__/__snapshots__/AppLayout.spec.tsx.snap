// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout /> Snapshot renders snapshot with no props 1`] = `ReactWrapper {}`;
